pyspark_ai/prompt.py

```diff
@@ -51,28 +64,49 @@
 Thought: Spark SQL does not support dynamic pivot operations, which are required to transpose the table as requested. I should get all the distinct values of column country.
 Action: query_sql_db
-Action Input: "SELECT DISTINCT Country FROM spark_ai_temp_view_93bcf0"
+Action Input: "SELECT DISTINCT `Country` FROM spark_ai_temp_view_93bcf0"
 Observation: USA, Canada, Mexico, China
 Thought: I can write a query to pivot the table by country and sum the amount for each fruit and country combination.
 Action: query_validation
-Action Input: SELECT * FROM spark_ai_temp_view_93bcf0 PIVOT (SUM(Amount) FOR Country IN ('USA', 'Canada', 'Mexico', 'China'))
+Action Input: SELECT * FROM spark_ai_temp_view_93bcf0 PIVOT (SUM(Amount) FOR `Country` IN ('USA', 'Canada', 'Mexico', 'China'))
 Observation: OK
 Thought:I now know the final answer.
-Final Answer: SELECT * FROM spark_ai_temp_view_93bcf0 PIVOT (SUM(Amount) FOR Country IN ('USA', 'Canada', 'Mexico', 'China'))"""
+Final Answer: SELECT * FROM spark_ai_temp_view_93bcf0 PIVOT (SUM(Amount) FOR `Country` IN ('USA', 'Canada', 'Mexico', 'China'))"""
     """QUESTION: Given a Spark temp view `spark_ai_temp_view_19acs2` with the following columns:
 ```
-Car STRING
+Car Name STRING
 Color STRING
 Make STRING
 ```
-Write a Spark SQL query to retrieve from view `spark_ai_temp_view_19acs2`: If color is gold, what is the total number of cars?
-Thought: I will use the column 'Color' to filter the rows where its value is 'gold' and then select the COUNT(`Car`) 
-because COUNT gives me the total number of cars.
+Write a Spark SQL query to retrieve from view `spark_ai_temp_view_19acs2`: What is the total number of cars that are gold?
+Thought: I should filter on the `Color` column to find the gold cars.
+I will now use the tool similar_value to help me choose my filter value.
+Action: similar_value
+Action Input: gold|Color|spark_ai_temp_view_19acs2
+Observation: gold
+Thought: The correct Color filter should be 'gold' because it is semantically closest to the keyword. I will use this in my query.
+I will use the column 'Color' to filter the rows where its value is 'gold' and then select the COUNT(`Car Name`) because COUNT gives me the total number of cars.
 Action: query_validation
-Action Input: SELECT COUNT(`Car`) FROM spark_ai_temp_view_19acs2 WHERE `Color` = 'gold'
+Action Input: SELECT COUNT(`Car Name`) FROM spark_ai_temp_view_19acs2 WHERE `Color` = 'gold'
 Observation: OK
 Thought: I now know the final answer.
-Final Answer: SELECT COUNT(`Car`) FROM spark_ai_temp_view_19acs2 WHERE `Color` = 'gold'"""
+Final Answer: SELECT COUNT(`Car Name`) FROM spark_ai_temp_view_19acs2 WHERE `Color` = 'gold'"""
+    """QUESTION: Given a Spark temp view `spark_ai_temp_view_19acs2` with the following columns:
+```
+Student STRING
+Birthday STRING
+```
+Write a Spark SQL query to retrieve from view `spark_ai_temp_view_12qcl3`: Who is the student with the birthday January 1, 2006?
```

Comment:
Let's make this question "What is the total number of students with the birthday January 1, 2006?"
and remove the example above.
The fewer examples, the lower the cost it is. These two seems able to be combined.